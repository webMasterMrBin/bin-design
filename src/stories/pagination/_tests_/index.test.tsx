import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('snapshot test', () => {
  it('pagination snapshot', () => {
    expect(toJson(render(<Pagination total={50} />))).toMatchSnapshot();
  });

  it('pagination test', () => {
    const wrapper = mount(<Pagination />);
    // 默认显示1-4页码
    //expect(nodeText.slice(0, 4)).toEqual(['1', '2', '3', '4']);
    // 点击最后一个item 即默认的第20页面
    wrapper
      .find('.pageitem-test')
      .last()
      .simulate('click');
    const nodeText = wrapper.find('.pageitem-test').map(o => o.text());
    expect(nodeText.slice(0, 5)).toEqual(['1', '17', '18', '19', '20']);
  });
});
