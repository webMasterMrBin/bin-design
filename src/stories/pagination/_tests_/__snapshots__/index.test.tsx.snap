// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test basic use 1`] = `
<div
  class="sc-htpNat ezYycK"
>
  <button
    class="sc-bdVaJa cgEJQX"
    disabled=""
  >
    &lt;
  </button>
  <div
    class="active sc-bwzfXH kaLEQY"
  >
    1
  </div>
  <div
    class="sc-bwzfXH kaLEQY"
  >
    2
  </div>
  <div
    class="sc-bwzfXH kaLEQY"
  >
    3
  </div>
  <div
    class="sc-bwzfXH kaLEQY"
  >
    4
  </div>
  <div>
    &gt;&gt;
  </div>
  <div
    class="sc-bwzfXH kaLEQY"
  >
    20
  </div>
  <button
    class="sc-bdVaJa cgEJQX"
  >
    &gt;
  </button>
  <div
    class="sc-bxivhb pJtMe"
  >
    跳转至
    <input
      type="text"
    >
      
    </input>
  </div>
</div>
`;

exports[`snapshot test pagination snapshot 1`] = `
<div
  class="sc-htpNat ezYycK"
>
  <button
    class="sc-bdVaJa cgEJQX"
    disabled=""
  >
    &lt;
  </button>
  <div
    class="active sc-bwzfXH pageitem-test kaLEQY"
  >
    1
  </div>
  <div
    class="sc-bwzfXH pageitem-test kaLEQY"
  >
    2
  </div>
  <div
    class="sc-bwzfXH pageitem-test kaLEQY"
  >
    3
  </div>
  <div
    class="sc-bwzfXH pageitem-test kaLEQY"
  >
    4
  </div>
  <div>
    &gt;&gt;
  </div>
  <div
    class="sc-bwzfXH pageitem-test kaLEQY"
  >
    10
  </div>
  <button
    class="sc-bdVaJa cgEJQX"
  >
    &gt;
  </button>
  <div
    class="sc-bxivhb pJtMe"
  >
    跳转至
    <input
      type="text"
    >
      
    </input>
  </div>
</div>
`;
